WordPress/Classes/ViewRelated/Post/EditPostViewController.swift

```diff
@@ -0,0 +1,293 @@
+    private init(post: Post?, blog: Blog?) {
+        if let post = post {
+            self.post = post
+        }
+        super.init(nibName: nil, bundle: nil)
+        if let blog = blog {
+            self.blog = blog
+        }
+        super.modalPresentationStyle = .OverCurrentContext
+        self.restorationIdentifier = RestorationKey.viewController.rawValue
+        self.restorationClass = EditPostViewController.self
+
+        addChildViewController(postPost)
+        view.addSubview(postPost.view)
+        postPost.didMoveToParentViewController(self)
+    }
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
+
+    override func viewWillAppear(animated: Bool) {
+        // show postpost, which will be transparent
+        postPost.modalPresentationStyle = .FullScreen
+        postPost.modalTransitionStyle = .CoverVertical
+        self.view.opaque = false
+        self.view.backgroundColor = UIColor.clearColor()
+
+        postPost.view.topAnchor.constraintEqualToAnchor(view.topAnchor)
+        postPost.view.bottomAnchor.constraintEqualToAnchor(view.bottomAnchor)
+        postPost.view.leftAnchor.constraintEqualToAnchor(view.leftAnchor)
+        postPost.view.rightAnchor.constraintEqualToAnchor(view.rightAnchor)
+    }
+
+    override func viewDidAppear(animated: Bool) {
+        if (!hasShownEditor) {
+            self.showEditor()
+            self.hasShownEditor = true
+        }
+    }
+
+    override func preferredStatusBarStyle() -> UIStatusBarStyle {
+        return .LightContent
+    }
+
+    // MARK: show the editor
+
+    private func showEditor() {
+        let editorSettings = EditorSettings()
+        let editor: UIViewController
+        if editorSettings.visualEditorEnabled {
+            if editorSettings.nativeEditorEnabled {
+                editor = editPostInNativeEditor()
+            } else {
+                editor = editPostInNewEditor()
+            }
+        } else {
+            editor = editPostInOldEditor()
+        }
+
+        postPost.presentViewController(editor, animated: !showImmediately) {
+            let generator = WPImpactFeedbackGenerator(style: .Medium)
+            generator.impactOccurred()
+        }
+    }
+
+    private func editPostInNativeEditor() -> UIViewController {
+        let postToEdit: Post
+        if let post = self.post {
+            postToEdit = post
+        } else {
+            let context = ContextManager.sharedInstance().mainContext
+            let postService = PostService(managedObjectContext: context)
+            postToEdit = postService.createDraftPostForBlog(blog)
+            WPAppAnalytics.track(.EditorCreatedPost, withProperties: ["tap_source": "posts_view"], withBlog: blog)
+        }
+
+        let postViewController = AztecPostViewController(post: postToEdit)
+        let navController = UINavigationController(rootViewController: postViewController)
+        navController.modalPresentationStyle = .FullScreen
```

Comment:
Should these methods be using the `editorModalPresentationStyle` instead of specifying `.FullScreen`? Otherwise it seems that we never use that property, and this overrides what all of the presenting view controllers use...